pal_event_server/lib/src/loggers/app_usage/idle_time.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+
+import 'package:logging/logging.dart';
+
+final _logger = Logger('IdleTime');
+
+RegExp pattern = RegExp(r'"HIDIdleTime" = (\d+)');
+
+Future<double> getIdleSecondsMacOS() async {
+  final result = await Process.run('ioreg', ['-rc' 'IOHIDSystem']);
+  final match = pattern.firstMatch(result.stdout);
+  if (match == null || match[1] == null) {
```

Comment:
Do we need to check for an out of bounds access on the match[1] prior to accessing it?